tools/rostest/src/rostest/rostest_parent.py

```diff
@@ -48,14 +49,46 @@
 
 class ROSTestLaunchParent(roslaunch.parent.ROSLaunchParent):
 
-    def __init__(self, config, roslaunch_files, port=0):
+    def __init__(self, config, roslaunch_files, port=0, reuse_master=False, clear=False):
         if config is None:
             raise Exception("config not initialized")
         # we generate a run_id for each test
-        run_id = roslaunch.core.generate_run_id()
+        if reuse_master:
+            param_server = rosgraph.Master('/roslaunch')
+            have_master = False
+            try:
+                run_id = param_server.getParam('/run_id')
+                have_master = True
+            except Exception as e:
+                rospy.logwarn("Could not connect to existing ROS master; will create"
```

Comment:
`rospy.logwarn` has not been imported in this file and fails at runtime.

Same below.